ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/commitmarkable/integration/CMRIntegrationTest.java

```diff
@@ -169,60 +167,40 @@ public void run() {
 									.createStatement();
 							createStatement
 									.execute("INSERT INTO foo (bar) VALUES (1)");
-							// System.out.printf("XXX txn close%n");
-
-							if (faultType == 1)
-								Runtime.getRuntime().halt(0);
 
                             userTransaction.commit();
 							connection.close(); // This wouldn't work for a
 												// none-JCA code as commit has
 												// closed the connection - it
 												// helps us though as JCA seems
 												// to rely on finalize
-
-							// System.out
-							// .printf("committed txn iteration %d%n", i);
 							success++;
 						} catch (SQLException e) {
-							System.err.println("boom");
-							e.printStackTrace();
-							if (e.getCause() != null) {
-								e.getCause().printStackTrace();
-							}
+							log.errorf(e, "Error while invoking insertion to a database table with CMR resource");
 							SQLException nextException = e.getNextException();
 							while (nextException != null) {
-								nextException.printStackTrace();
-								nextException = nextException
-										.getNextException();
-							}
-							Throwable[] suppressed = e.getSuppressed();
-							for (int j = 0; j < suppressed.length; j++) {
-								suppressed[j].printStackTrace();
```

Comment:
Why aren't we printing the stack trace for suppressed exceptions?

Reply:
That's done by logger automatically. It prints nicely cause exceptions and the suppressed exceptions without this manual handling. The logger is placed clearly with the level of severity and indentation under the `standalone.log` (easier to read, easier to understand the cause chain with the suppressed exception printed in the hierarchy).
That's all the reasons for these changes.